go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -54,26 +56,69 @@ func podToCookie(pod *kapi.Pod) string {
 	return nameToCookie(pod.Namespace + "_" + pod.Name)
 }
 
-func (n *NodeController) addOrUpdatePod(pod *kapi.Pod) error {
+func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, wf *factory.WatchFactory) error {
 	podIPs, podMAC, err := getPodDetails(pod, n.nodeName)
 	if err != nil {
 		klog.V(5).Infof("cleaning up hybrid overlay pod %s/%s because %v", pod.Namespace, pod.Name, err)
 		return n.deletePod(pod)
 	}
 
+	namespace, err := ovn.WaitForNamespace(pod.Namespace, wf)
+	if err != nil {
+		return err
+	}
+	namespaceExternalGW := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
 	cookie := podToCookie(pod)
 	for _, podIP := range podIPs {
 		_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
 			fmt.Sprintf("table=10,cookie=0x%s,priority=100,ip,nw_dst=%s,actions=set_field:%s->eth_src,set_field:%s->eth_dst,output:ext", cookie, podIP.IP, n.drMAC, podMAC))
 		if err != nil {
 			return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
 		}
+		if namespaceExternalGW != "" {
+			_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
+				fmt.Sprintf("table=0, cookie=0x%s, priority=100, ip, nw_src=%s, actions=load:%s->NXM_NX_TUN_ID[0..31],set_field:%s->tun_dst,output:%s", cookie, podIP.IP, strconv.Itoa(hotypes.HybridOverlayVNI), namespaceExternalGW, extVXLANName))
+			if err != nil {
+				return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+			}
+			node, err := wf.GetNode(n.nodeName)
+			if err != nil {
+				return fmt.Errorf("failed to get node %s: %v", n.nodeName, err)
+			}
+			subnet, _, _, err := getNodeDetails(node)
+			if err != nil {
+				return fmt.Errorf("failed to get node %s details: %v", n.nodeName, err)
+			}
+			extBridgeMacAddress, err := util.GetOVSPortMACAddress(extBridgeName)
+			if err != nil {
+				return fmt.Errorf("failed to get external bridge %s: %v", extBridgeName, err)
+			}
+			extBridgeMacRaw := strings.Replace(extBridgeMacAddress.String(), ":", "", -1)
```

Comment:
@rcarrillocruz Also, can't we just do this in `ensureHybridOverlayBridge()` since it's not going to change over the life of the node?

Reply:
if we don't care about having this flow even if hybrid external gw is not used I'm fine with it

Reply:
Windows doesn't support ARP over vxlan anyway IIRC, which is why DRMAC exists.